syntax = "proto3";

package GrpcApiPractice;

import "Models/Product.proto";
import "Models/ServiceStatus.proto";
import "google/protobuf/empty.proto";

service ProductService
{
    rpc CreateProduct(GrpcApiPractice.Product) returns (GrpcApiPractice.ServiceStatus) { }
    rpc ListProduct(google.protobuf.Empty) returns (ListProductResponse) { }
    rpc ListProductById (ProductByIdRequest) returns (ListProductResponse) { }
}

message ProductByIdRequest
{
    int64 product_id = 1;
}

message ListProductResponse
{
  repeated GrpcApiPractice.Product products = 1;
}